sdk/azcore/internal/exported/response_error.go

```diff
@@ -117,7 +117,7 @@ type ResponseError struct {
 	StatusCode int
 
 	// RawResponse is the underlying HTTP response.
-	RawResponse *http.Response
+	RawResponse *http.Response `json:"-"`
```

Comment:
Doesn't this hinder debugability?

Please add more to comment to explain this change and link back to the issue. In a year the context for why we do this will be vague.

Reply:
Say I call some API and it fails, returning an `*azcore.ResponseError`. I then want to marshal that error so I can send it somewhere else.
```go
bytes, err := json.Marshal(someRespErr)
```
This will fail due to fields in `RawResponse.Request` that cannot be marshaled (func types).
This change tells the JSON marshaler (and unmarshaler) to skip the `RawResponse` field.

Reply:
Oh, if you mean debuggability on the client side then the answer is yes as the error details will be lost. That said, we've never considered the contents returned by `Error()` to be contractual. Whether or not this falls into that bucket is debatable.

We can probably preserve the error message, but it will require computing the message, stashing it in an unexported field, and adding custom JSON un/marshaler methods that know how to handle it. We'd still need to omit the `RawResponse` though.

Reply:
I've updated the marshaler to preserve the contents of the error message and added a doc comment describing the behavior.